pkg/scheduler/framework/plugins/dynamicresources/dra_manager.go

```diff
@@ -263,3 +294,42 @@ func (c *claimTracker) AssumeClaimAfterAPICall(claim *resourceapi.ResourceClaim)
 func (c *claimTracker) AssumedClaimRestore(namespace, claimName string) {
 	c.cache.Restore(namespace + "/" + claimName)
 }
+
+// deviceClassEventHandler handles device class informer events to update the extended resource cache.
+type deviceClassEventHandler struct {
```

Comment:
Instead of introducing a new type just for registering the event handlers, can we make `ExtendedResourceCache` implement these event handlers: OnAdd, OnUpdate and OnDelete? I think the abstractions will be simpler that way